diff -up gnu-efi-3.0.18/apps/Makefile.3~ gnu-efi-3.0.18/apps/Makefile
--- gnu-efi-3.0.18/apps/Makefile.3~	2024-07-06 16:49:15.898399072 +0200
+++ gnu-efi-3.0.18/apps/Makefile	2024-07-06 16:50:01.214897473 +0200
@@ -51,7 +51,7 @@ ifneq (,$(findstring FreeBSD,$(OS)))
 LDSCRIPT	= $(TOPDIR)/gnuefi/elf_$(ARCH)_fbsd_efi.lds
 endif
 
-LDFLAGS		+= -shared -Bsymbolic -L$(TOPDIR)/$(ARCH)/lib -L$(TOPDIR)/$(ARCH)/gnuefi $(CRTOBJS)
+LDFLAGS		+= -shared -Bsymbolic -z noexecstack --no-warn-rwx-segments -L$(TOPDIR)/$(ARCH)/lib -L$(TOPDIR)/$(ARCH)/gnuefi $(CRTOBJS)
 
 LOADLIBES	+= -lefi -lgnuefi
 LOADLIBES	+= $(LIBGCC)
